In [223]:
import pandas as pd

In [224]:
# Lendo arquivo csv com pandas

df=pd.read_csv('dataset_atividade_rede_neural_industrial.csv')

print(df.head())
print(df.shape)

   temperatura  vibracao  pressao  corrente_motor  tempo_operacao  falha
0        68.98      3.91    23.61           13.04             122      0
1        62.25      2.86    30.21           18.46             164      0
2        74.18      2.99    34.16           15.41              91      0
3        71.91      3.65    27.44           19.80             203      0
4        82.15      3.59    30.18           17.05             234      1
(150, 6)


In [225]:
# Estatística descritiva dos dados

print(df.describe())

       temperatura    vibracao     pressao  corrente_motor  tempo_operacao  \
count   150.000000  150.000000  150.000000      150.000000      150.000000   
mean     72.617800    3.330933   30.157933       15.089667      168.460000   
std       6.730211    0.851747    3.985320        3.241509       39.674621   
min      49.090000    1.430000   20.330000        7.230000       46.000000   
25%      68.710000    2.800000   27.377500       13.115000      144.250000   
50%      72.950000    3.340000   30.345000       14.960000      164.000000   
75%      76.702500    3.967500   32.775000       17.255000      196.000000   
max      91.110000    5.260000   40.800000       23.890000      273.000000   

            falha  
count  150.000000  
mean     0.313333  
std      0.465403  
min      0.000000  
25%      0.000000  
50%      0.000000  
75%      1.000000  
max      1.000000  


# Preparação dos dados

In [226]:

# Verificando a existência de dados nulos

print(df.isnull().sum())

temperatura       0
vibracao          0
pressao           0
corrente_motor    0
tempo_operacao    0
falha             0
dtype: int64


In [227]:
# Verificando a existência de dados duplicados

print(df.duplicated().sum())

0


# Criando uma rede Neuram com Sklearn

In [228]:
# Importando as bibliotecas


from sklearn.model_selection import train_test_split
from sklearn.neural_network import MLPClassifier
from sklearn.preprocessing import StandardScaler
from sklearn.metrics import accuracy_score

In [229]:
# Definindo as variáveis

x=df.drop('falha',axis=1)
y=df['falha']

In [230]:
# Separando os dados de treinamento e teste

x_train,x_test,y_train,y_test=train_test_split(x,y,test_size=0.3,random_state=42)

In [231]:
scaler = StandardScaler()
# Ajusta o escalonador nos dados de treino e transforma
x_train_scaled = scaler.fit_transform(x_train)
# Apenas transforma os dados de teste (sem ajustar novamente)
x_test_scaled = scaler.transform(x_test)

In [232]:
# Criando o modelo

model=MLPClassifier(hidden_layer_sizes=(5,),activation='relu',max_iter=100000,random_state=42)

In [233]:
# Treinando o modelo

In [234]:
model.fit(x_train_scaled,y_train)

MLPClassifier(hidden_layer_sizes=(5,), max_iter=100000, random_state=42)

In [235]:
# Calculando a acurácia

y_pred=model.predict(x_test_scaled)
accuracy=accuracy_score(y_test,y_pred)
print('Acurácia:',accuracy)

Acurácia: 0.8888888888888888


# Relatório

1. Contexto do Experimento: O objetivo deste trabalho foi desenvolver e avaliar uma Rede Neural Artificial (Multilayer Perceptron - MLP) para prever falhas em componentes industriais. O conjunto de dados original conta com 150 amostras tendo como variáveis: temperatura, vibração, pressão, corrente do motor, tempo de operação e falha.

2. Configuração da Rede Neural: MLPClassifier (Scikit-Learn)
Arquitetura: 1 camada oculta com 5 neurônios
Função de Ativação: ReLU  
Divisão dos Dados: 70% para treinamento e 30% para testes

3. Resultado: O modelo atingiu uma acurácia de aproximadamente 88,89% nos dados de teste. Em termos práticos, isso significa que a cada 100 diagnósticos que a rede neural faz sobre as máquinas da fábrica, ela acerta cerca de 89 deles (identificando corretamente tanto os casos em que a máquina opera normalmente quanto os casos em que ela apresenta uma falha). Uma acurácia de ~89% indica que as variáveis coletadas possuem forte correlação com o estado de falha, permitindo que o algoritmo aprenda os padrões operacionais com eficiência
